### PR TITLE
Update dependencies pre-v5

### PR DESCRIPTION
**Pull request by emilbjorklund**
Updated as many dependencies that I could, in preparation of a V5 (where we can drop baseApp-plugin, and add support for webpack 5 + multi-compiler mode etc).

* Bumped minimum node version to 14
* Left some typescript dependencies out of it, since something was broken w. `clean-webpack-plugin` etc.
* Updated to new v7 API of Husky, no more `package.json` config. 
* Since some JS used in test bundling was updated due to new Babel and related polyfills, test hashes had to be updated.
* Some TS definitions from webpack-manifest-plugins updated/simplified in their v3.

### CODE REVIEW

File: .husky/pre-push [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (emilbjorklund) [inline .husky/pre-push:1]: This is the new format for husky setup - no config file or sections in package.json, just small files with the command in them.

File: package.json [hunk 2]
```diff
@@ -39,15 +39,11 @@
...
     "test:mocha": "NODE_ENV=test nyc --reporter=lcov --reporter text mocha --reporter dot test/**/*.spec.js",
     "test-checkless": "run-p test:*",
-    "test": "run-p check && run-p test:*"
-  },
-  "husky": {
-    "hooks": {
-      "pre-push": "npm test"
-    }
+    "test": "run-p check && run-p test:*",
+    "postinstall": "husky install"
```
Comment (emilbjorklund) [inline package.json:43]: Added a `postinstall` command, per recommendation of husky docs. Should run automatically upon installation.

File: example/example-asset-versions-sourcemaps.json [hunk 3]
```diff
@@ -12,8 +12,8 @@
 		"runtime.7845eec1df183fbac714.js.map": "dist/rev/runtime.7845eec1df183fbac714.js.map",
 		"lazy.9baa1f3abc282ca225f0.js": "dist/rev/lazy.9baa1f3abc282ca225f0.js",
 		"lazy.9baa1f3abc282ca225f0.js.map": "dist/rev/lazy.9baa1f3abc282ca225f0.js.map",
-		"legacy.83b8f97ff2272c92b172.js": "dist/rev/legacy.83b8f97ff2272c92b172.js",
-		"legacy.83b8f97ff2272c92b172.js.map": "dist/rev/legacy.83b8f97ff2272c92b172.js.map"
+		"legacy.ec898fcde7adf983517f.js": "dist/rev/legacy.ec898fcde7adf983517f.js",
+		"legacy.ec898fcde7adf983517f.js.map": "dist/rev/legacy.ec898fcde7adf983517f.js.map"
```
Comment (emilbjorklund) [inline example/example-asset-versions-sourcemaps.json:16]: Had to change a bunch of file hashes due to new Babel, new core-js etc.

File: example/webpack.config.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ const path = require('path');
 const webpack = require('webpack');
 
 const { CleanWebpackPlugin } = require('clean-webpack-plugin');
-const ManifestPlugin = require('webpack-manifest-plugin');
+const { WebpackManifestPlugin } = require('webpack-manifest-plugin');
```
Comment (emilbjorklund) [inline example/webpack.config.js:11]: New import structure for webpack-manifest-plugin

**Review by morreman — COMMENTED** [review]
Looks good to me 🚀

**Pull request merged**